input-forms.md

```diff
@@ -0,0 +1,82 @@
+# Input-forms Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**/api/configuration/input-forms**   
+
+Provide access to the configured input-forms. It returns the list of existent input-forms.
+Please note that starting from DSpace 7 each input-form will describe a single page of inputs, combining different pages togheter is done with the [submission-definition](submission-panels.md), aka the item-submission.xml configuration file 
+
+Example: to be provided
+
+## Single Input-Form 
+**/api/configuration/input-forms/<:form-name>**
+
+Provide detailed information about a specific input-form. The JSON response document is as follow
+```json
+{
+  "name": "traditional",
+  "fields": [
+  		{
+  			label: "Authors",
+  			repeatable: false,
+  			mandatory: null,
+  			hints: "Enter the names of the authors of this item.",
+  			input: {
+  				type: "name"
+  			},
+  			selectableMetadata: [
+  				{
+  					"metadata": "dc.contributor.author",
+  					"authority": "SolrAuthorAuthority",
+  					"closed": false
+  				}
+  			],
+  			scope: null, 
+  			visibility: {
+  				main: null,
+  				other: null
+  			},
+  			typeBind: [],
+  			languageCodes: []
+  		},
+  		{
+			label: "Title",
+  			repeatable: false,
+  			mandatory: "You must enter a main title for this item.",
+  			hints: "Enter the main title of the item.",
+  			input: {
+  				type: "onebox",
+  				regex: null
+  			},
+  			selectableMetadata: [
+  				{
+  					metadata: "dc.title"
+  				}
+  			],
+  			scope: null, 
+  			visibility: {
+  				main: null,
+  				other: null
+  			},
+  			typeBind: [],
+  			languageCodes: [
```

Comment:
Currently you cannot specify "selectable" languages in the input-forms.xml file. Does this then relate to the enabled locales in dspace.cfg or do you plan to add it to the input-forms.xml file as a new XML element?

Reply:
In the current version the list of languages that apply to multilingual metadata is configured in the value-pairs named common_iso_languages see
https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/app/util/DCInput.java#L36
https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/app/util/DCInput.java#L119

as it is a common need to allow different list of languages for some metadata we want to have a contract more general and specify this list on a per metadata basis. The functionality itself will be not implemented right now but this give as the freedom in future to do that without broking the REST contract (and the UI).